preprocessors/multistage-textbook-diagram.schema.json

```diff
@@ -0,0 +1,149 @@
+                        "items": {
+                            "type": "object",
+                            "description": "Contains information about each individual segment in the stage. Information such as area, centre of segment etc are included in this section",
+                            "properties": {
+                                "name": {
+                                    "description": "The name of the segment. Provides additional detail regarding subsections of a stage, if present.",
+                                    "type": "string"
+                                },
+                                "contours": {
+                                    "description": "The separate contours making up the segment.",
+                                    "type": "array",
+                                    "items": {
+                                        "$ref": "#/definitions/contour"
+                                    }
+                                },
+                                "centroid": {
+                                    "allOf": [
+                                        {
+                                            "$ref": "#/definitions/normCoord"
+                                        },
+                                        {
+                                            "description": "The overall centroid of the segment with normalized coordinates."
+                                        }
+                                    ]
+                                },
+                                "area": {
+                                    "description": "Total area occupied by the segment in a given stage. the area is normalised between 0 and 1",
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                }
+                            },
+                            "required": [
+                                "contours",
+                                "centroid",
+                                "area"
+                            ]
+                        }
+                    },
+                    "id": {
+                        "type": "string",
+                        "description": "Unique identifier for the stage"
+                    },
+                    "label": {
+                        "type": "string",
+                        "description": "Name of the stage"
+                    },
+                    "description": {
+                        "type": "string",
+                        "description": "One/ two sentence description of the stage"
+                    }
+                },
+                "required": [
+                    "id",
+                    "label",
+                    "segments"
+                ]
+            }
+        },
+        "links": {
```

Comment:
What are links?

Reply:
Links are the connections between the different stages. Will add a description for the same.